std/json.ln

```diff
@@ -532,16 +535,29 @@ export fn toString(j: JSON): string {
     };
   }).toHashMap();
   // Construct a list of nodes to process, based on the parents of the leaf nodes, using a HashMap
-  // as a set
+  // as a set, but only those parents that have all children set by these leaves
   let nodeList = newHashMap(-1, false);
   leaves.eachLin(fn (l: Node<JSONNode>) {
     const parentId = l.getParent().id;
     if parentId > -1 {
       nodeList.set(l.getParent().id, true);
     }
   });
+  nodeList.keyVal().filter(fn (kv: KeyVal<int64, bool>): bool {
+    const n = j.getNodeById(kv.key);
+    const children = n.getChildren();
+    let missingChildren = false;
+    children.eachLin(fn (c: Node<JSONNode>) {
+      if lookup.get(c.id).isErr() {
+        missingChildren = true;
+      }
+    });
+    return missingChildren;
+  }).eachLin(fn (kv: KeyVal<int64, bool>) {
+    nodeList.set(kv.key, false);
+  });
   // Execute the following block over and over again until there are no more nodes to process
-  seqdo(newseq(pow(2, 64).getOr(0)), fn (): bool {
+  seqdo(newseq(pow(2, 63).getOr(0)), fn (): bool {
```

Comment:
what's this about?

Reply:
Oh, the actual 2^64 doesn't fit in 64-bits, so it was hitting the `.getOr(0)` and then never running more than once for me.

Reply:
wow, that is a fun bug. we might have other occurrences of `pow(2, 64)` in `root.ln` for the tree implementation no?

Reply:
You're right, there is one other place this exists, I will patch it.